### PR TITLE
:seedling: Add network builtins to topology controller

### LINKED ISSUE #5578: ClusterClass builtin variables: ClusterNetwork
The current [ClusterClass proposal](https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/202105256-cluster-class-and-managed-topologies.md#builtin-variables) does not include any builtin variables for `Cluster.spec.clusterNetwork`. This means it's not possible to access any of those fields in ClusterClass patches.

Let's collect use cases (if there are any) for those fields, so that we can add builtin variables accordingly if there is demand.

`spec.clusterNetwork` contains the following fields:

```go
type ClusterNetwork struct {
	// APIServerPort specifies the port the API Server should bind to.
	// Defaults to 6443.
	// +optional
	APIServerPort *int32 `json:"apiServerPort,omitempty"`

	// The network ranges from which service VIPs are allocated.
	// +optional
	Services *NetworkRanges `json:"services,omitempty"`

	// The network ranges from which Pod networks are allocated.
	// +optional
	Pods *NetworkRanges `json:"pods,omitempty"`

	// Domain name for services.
	// +optional
	ServiceDomain string `json:"serviceDomain,omitempty"`
}
```

/kind feature
/area topology

### PR DESCRIPTION
**Pull request by killianmuldoon**
This PR adds a new set of builtin variables from the Cluster's `.spec.clusterNetwork` field. 

Fixes #5578

### CODE REVIEW

File: internal/controllers/topology/cluster/patches/variables/variables.go [hunk 1]
```diff
@@ -64,6 +67,20 @@ type ClusterTopologyBuiltins struct {
 	Class string `json:"class,omitempty"`
 }
 
+// ClusterNetworkBuiltins represents builtin cluster network variables.
+type ClusterNetworkBuiltins struct {
+	// APIServerPort is the port the API Server should bind to.
+	APIServerPort *int32 `json:"apiServerPort,omitempty"`
+	// 	ServiceDomain is the network ranges from which service VIPs are allocated.
+	ServiceDomain string `json:"serviceDomain,omitempty"`
```
Comment (sbueringer) [inline internal/controllers/topology/cluster/patches/variables/variables.go:75]:
```suggestion
	ServiceDomain *string `json:"serviceDomain,omitempty"`
```
I think this should be a pointer too. Given that the Cluster variable is also option (don't ask me why the serviceDomain in the Cluster is not a pointer :) )

File: internal/controllers/topology/cluster/patches/variables/variables.go [hunk 2]
```diff
@@ -64,6 +67,20 @@ type ClusterTopologyBuiltins struct {
 	Class string `json:"class,omitempty"`
 }
 
+// ClusterNetworkBuiltins represents builtin cluster network variables.
+type ClusterNetworkBuiltins struct {
+	// APIServerPort is the port the API Server should bind to.
+	APIServerPort *int32 `json:"apiServerPort,omitempty"`
+	// 	ServiceDomain is the network ranges from which service VIPs are allocated.
```
Comment (sbueringer) [inline internal/controllers/topology/cluster/patches/variables/variables.go:74]:
```suggestion
	// ServiceDomain is the domain name for services.
```

File: internal/controllers/topology/cluster/patches/variables/variables.go [hunk 3]
```diff
@@ -64,6 +67,20 @@ type ClusterTopologyBuiltins struct {
 	Class string `json:"class,omitempty"`
 }
 
+// ClusterNetworkBuiltins represents builtin cluster network variables.
+type ClusterNetworkBuiltins struct {
+	// APIServerPort is the port the API Server should bind to.
+	APIServerPort *int32 `json:"apiServerPort,omitempty"`
```
Comment (sbueringer) [inline internal/controllers/topology/cluster/patches/variables/variables.go:73]: @fabriziopandini Given the ongoing discussion around APIServer local/lb ports. Do you think it's safe to provide this as a builtin variable? (or another point of view, if we don't have demand for it, maybe we should omit it given that we might want to change the API soon)

File: internal/controllers/topology/cluster/patches/variables/variables.go [hunk 4]
```diff
@@ -64,6 +67,20 @@ type ClusterTopologyBuiltins struct {
...
+	// 	ServiceDomain is the network ranges from which service VIPs are allocated.
+	ServiceDomain string `json:"serviceDomain,omitempty"`
+	// ServicesCIDRBlocks is network ranges from which Service networks are allocated.
+	ServicesCIDRBlocks []string `json:"services,omitempty"`
```
Comment (sbueringer) [inline internal/controllers/topology/cluster/patches/variables/variables.go:77]:
```suggestion
	// ServiceCIDRs are the network ranges from which Service VIPs are allocated.
	ServiceCIDRs []string `json:"serviceCIDRs,omitempty"`
```

File: internal/controllers/topology/cluster/patches/variables/variables.go [hunk 5]
```diff
@@ -64,6 +67,20 @@ type ClusterTopologyBuiltins struct {
...
+	// PodsCIDRBlocks is network ranges from which Pod networks are allocated.
+	PodsCIDRBlocks []string `json:"pods,omitempty"`
```
Comment (sbueringer) [inline internal/controllers/topology/cluster/patches/variables/variables.go:79]:
```suggestion
	// PodCIDRs are the network ranges from which Pod networks are allocated.
	PodCIDRs []string `json:"podCIDRs,omitempty"`
```

File: internal/webhooks/patch_validation.go [hunk 6]
```diff
@@ -335,6 +335,14 @@ var builtinVariables = sets.NewString(
 	"builtin.cluster.topology.version",
 	"builtin.cluster.topology.class",
 
+	// ClusterNetwork builtins
```
Comment (sbueringer) [inline internal/webhooks/patch_validation.go:338]: Just double-checking. That was the only place we duplicated the builtins to?

File: internal/controllers/topology/cluster/patches/variables/variables.go [hunk 7]
```diff
@@ -118,6 +135,11 @@ func Global(clusterTopology *clusterv1.Topology, cluster *clusterv1.Cluster) (Va
 		variables[variable.Name] = variable.Value
 	}
 
+	clusterNetworkIPFamily, err := cluster.GetIPFamily()
+	if err != nil {
+		return variables, err
```
Comment (sbueringer) [inline internal/controllers/topology/cluster/patches/variables/variables.go:140]:
```suggestion
		return nit, err
```
Let's please return nil instead of a partial result

File: internal/controllers/topology/cluster/patches/variables/variables.go [hunk 8]
```diff
@@ -127,6 +149,13 @@ func Global(clusterTopology *clusterv1.Topology, cluster *clusterv1.Cluster) (Va
 				Version: cluster.Spec.Topology.Version,
 				Class:   cluster.Spec.Topology.Class,
 			},
+			Network: &ClusterNetworkBuiltins{
+				APIServerPort:      cluster.Spec.ClusterNetwork.APIServerPort,
+				ServiceDomain:      cluster.Spec.ClusterNetwork.ServiceDomain,
+				ServicesCIDRBlocks: cluster.Spec.ClusterNetwork.Services.CIDRBlocks,
+				PodsCIDRBlocks:     cluster.Spec.ClusterNetwork.Pods.CIDRBlocks,
```
Comment (sbueringer) [inline internal/controllers/topology/cluster/patches/variables/variables.go:156]: These two can panic.

EDIT: All others too as ClusterNetwork is a pointer

Comment (sbueringer) [reply internal/controllers/topology/cluster/patches/variables/variables.go:156]: Let's please also have unit tests to make sure we don't panic if ClusterNetwork or the service/pod CIDRs are nil

File: internal/controllers/topology/cluster/patches/engine_test.go [hunk 9]
```diff
@@ -476,6 +476,19 @@ func setupTestObjects() (*scope.ClusterBlueprint, *scope.ClusterState) {
 			Namespace: metav1.NamespaceDefault,
 		},
 		Spec: clusterv1.ClusterSpec{
+			Paused: false,
```
Comment (sbueringer) [inline internal/controllers/topology/cluster/patches/engine_test.go:479]: Can we add a patch to the "Should apply JSON patches to InfraClus.." test case above who uses one of those vars?

File: internal/controllers/topology/cluster/patches/variables/variables.go (hunk 3 above)
Comment (fabriziopandini) [reply internal/controllers/topology/cluster/patches/variables/variables.go:73]: @yastij opinions?
I don't see this going away soon, but at the same time holding back people from using it in patch makes sense if the plan is to deprecate it

File: internal/controllers/topology/cluster/patches/variables/variables.go (hunk 4 above)
Comment (sbueringer) [reply internal/controllers/topology/cluster/patches/variables/variables.go:77]: /cc @fabriziopandini @vincepri to get consensus on the variable name (same in https://github.com/kubernetes-sigs/cluster-api/pull/6171#discussion_r811348749)

**Comment by sbueringer** [comment]
@fabriziopandini Do we want to include this PR in the next 1.1.x release too, like we discussed yesterday for the new MD/CP builtin variables

File: internal/webhooks/patch_validation.go (hunk 6 above)
Comment (sbueringer) [reply internal/webhooks/patch_validation.go:338]: I'll resolve. Had to check myself too for my PR :)

File: internal/controllers/topology/cluster/patches/variables/variables.go (hunk 7 above)
Comment (sbueringer) [reply internal/controllers/topology/cluster/patches/variables/variables.go:140]: EDIT: `return nit, err` => `return nil, err` :joy:

File: internal/controllers/topology/cluster/patches/variables/variables_test.go [hunk 10]
```diff
@@ -66,12 +66,22 @@ func TestGlobal(t *testing.T) {
...
+						Pods: &clusterv1.NetworkRanges{
+							CIDRBlocks: []string{"11.10.10.1/24"},
+						},
+						ServiceDomain: "cluster.local",
+					},
 				},
 			},
 			want: VariableMap{
 				"location":   toJSON("\"us-central\""),
 				"cpu":        toJSON("8"),
```
Comment (sbueringer) [inline internal/controllers/topology/cluster/patches/variables/variables_test.go:83]: Just fyi. I will migrate some of the other unit tests here to use `toJSONCompact` in a PR I'll open today. Just in case you want to migrate the Cluster tests too. It's so much easier to read / check if it's correct.

File: internal/controllers/topology/cluster/patches/engine_test.go (hunk 9 above)
Comment (sbueringer) [reply internal/controllers/topology/cluster/patches/engine_test.go:479]: Just realized that this would have to be in json_patch_generator_test.go, but it's not really worth it.

File: internal/controllers/topology/cluster/patches/variables/variables_test.go (hunk 10 above)
Comment (fabriziopandini) [reply internal/controllers/topology/cluster/patches/variables/variables_test.go:83]: If we aim to backport, let's move all the test refactoring to a follow up PR

File: internal/controllers/topology/cluster/patches/variables/variables.go (hunk 3 above)
Comment (yastij) [reply internal/controllers/topology/cluster/patches/variables/variables.go:73]: I'd rather not expose it as a patch and have the ControlPlaneEndpoint + the new `LocalAPIServerPort` exposed. at least folks moving to clusterClass would be using the correct fields for port management

Comment (killianmuldoon) [reply internal/controllers/topology/cluster/patches/variables/variables.go:73]: I think we can leave it out of this PR and if it there's demand for it we can decide whether or not to add it in later.

File: internal/controllers/topology/cluster/patches/variables/variables_test.go (hunk 10 above)
Comment (killianmuldoon) [reply internal/controllers/topology/cluster/patches/variables/variables_test.go]: I've moved the cluster test to toJSONCompact - the only issue will be to rebase one of the variable-adding PRs on top of the other.

Comment (sbueringer) [reply internal/controllers/topology/cluster/patches/variables/variables_test.go]: Yup. I think we should be fine when we cherry-pick them in the same order.

File: internal/controllers/topology/cluster/patches/variables/variables.go [hunk 11]
```diff
@@ -64,6 +68,18 @@ type ClusterTopologyBuiltins struct {
 	Class string `json:"class,omitempty"`
 }
 
+// ClusterNetworkBuiltins represents builtin cluster network variables.
+type ClusterNetworkBuiltins struct {
+	// ServiceDomain is the domain name for services.
+	ServiceDomain *string `json:"serviceDomain,omitempty"`
+	// ServicesCIDRs is network ranges from which Service networks are allocated.
```
Comment (sbueringer) [inline internal/controllers/topology/cluster/patches/variables/variables.go:75]:
```suggestion
	// ServicesCIDRs are network ranges from which Service VIPs are allocated.
```
I think this should be really service VIPs. As far as I'm aware there are no service networks.

As far as I know:
* kube-controller-manager will use this range to hand out new service VIPs
* the PodsCIDRs are used to create new per-node "pod networks" == CIDRs (at least with some default Kubernetes CNI IPAM implementation)

File: internal/controllers/topology/cluster/patches/variables/variables.go [hunk 12]
```diff
@@ -64,6 +68,18 @@ type ClusterTopologyBuiltins struct {
...
+	ServicesCIDRs []string `json:"services,omitempty"`
```
Comment (sbueringer) [inline internal/controllers/topology/cluster/patches/variables/variables.go:76]:
```suggestion
	Services []string `json:"services,omitempty"`
```
Q: as we called the field `services` in JSON, should we call the field in the struct the same?
(same applies to pods)

File: internal/controllers/topology/cluster/patches/variables/variables_test.go [hunk 13]
```diff
@@ -66,12 +66,135 @@ func TestGlobal(t *testing.T) {
...
+  "namespace": "default",
+  "topology":{
+    "version": "v1.21.1",
+    "class": "clusterClass1"
+  },
+  "network":{
+    "serviceDomain":"cluster.local",
+    "services":["10.10.10.1/24"],
+    "pods":["11.10.10.1/24"],
+    "ipFamily": "IPv4IPFamily"
```
Comment (sbueringer) [inline internal/controllers/topology/cluster/patches/variables/variables_test.go:95]: Q: Should we make this "IPv4" instead? I understand why the go types have that suffix, but I think in general I think it's not good if the actual enum values have the enum type as suffix 
```suggestion
    "ipFamily": "IPv4"
```

File: internal/controllers/topology/cluster/patches/variables/variables_test.go [hunk 14]
```diff
@@ -66,12 +66,135 @@ func TestGlobal(t *testing.T) {
...
+					},
 				},
 			},
 			want: VariableMap{
-				"location":   toJSON("\"us-central\""),
-				"cpu":        toJSON("8"),
-				BuiltinsName: toJSON("{\"cluster\":{\"name\":\"cluster1\",\"namespace\":\"default\",\"topology\":{\"version\":\"v1.21.1\",\"class\":\"clusterClass1\"}}}"),
+				"location": toJSON("\"us-central\""),
+				"cpu":      toJSON("8"),
+				BuiltinsName: toJSONCompact(`{
```
Comment (sbueringer) [inline internal/controllers/topology/cluster/patches/variables/variables_test.go:83]: One additional ask, can you please indent the JSON values too? (like I did now on my PR after Fabrizio's finding)

Note: I used tabs there to make sure it's indented the same way as the regular Go code.

File: internal/controllers/topology/cluster/patches/variables/variables_test.go (hunk 13 above)
Comment (sbueringer) [reply internal/controllers/topology/cluster/patches/variables/variables_test.go:95]: On the other side that's what the IPFamily func returns, but I think it would be really not good to have those suffixes.


@vincepri @fabriziopandini Opinions?

File: internal/controllers/topology/cluster/patches/variables/variables.go (hunk 11 above)
Comment (killianmuldoon) [reply internal/controllers/topology/cluster/patches/variables/variables.go:75]: I've copied these comments from the API types which I think it's best to follow to reduce confusion. If we want to discuss the comments on the API types lets move it to a new issue.

File: internal/controllers/topology/cluster/patches/variables/variables_test.go (hunk 13 above)
Comment (killianmuldoon) [reply internal/controllers/topology/cluster/patches/variables/variables_test.go:95]: We're not doing any validation in this PR - these are the types defined in the api package under Cluster types. For the test I can change it to "foo" if we want. Is there need to do some specific evaluation or transformation of these types in this PR?

File: internal/controllers/topology/cluster/patches/variables/variables.go (hunk 11 above)
Comment (sbueringer) [reply internal/controllers/topology/cluster/patches/variables/variables.go]: Okay, kind of agree, but I'm probably missing something. Those are the comments on the API types:
```go
	// The network ranges from which service VIPs are allocated.
	// +optional
	Services *NetworkRanges `json:"services,omitempty"`

	// The network ranges from which Pod networks are allocated.
	// +optional
	Pods *NetworkRanges `json:"pods,omitempty"`
```

Comment (killianmuldoon) [reply internal/controllers/topology/cluster/patches/variables/variables.go]: Sorry - you're dead right. I'll update.

File: internal/controllers/topology/cluster/patches/variables/variables_test.go (hunk 13 above)
Comment (sbueringer) [reply internal/controllers/topology/cluster/patches/variables/variables_test.go]: Oh sorry for the confusion. I just added the comment at the test because it seemed convenient. I wanted to discuss to what values we should set that builtin variable.

File: internal/controllers/topology/cluster/patches/variables/variables.go (hunk 11 above)
Comment (sbueringer) [reply internal/controllers/topology/cluster/patches/variables/variables.go]: Happens :)

File: internal/controllers/topology/cluster/patches/variables/variables_test.go (hunk 13 above)
Comment (sbueringer) [reply internal/controllers/topology/cluster/patches/variables/variables_test.go]: If we want to change the return value of IPFamily func we could:
* set the builtin variables without suffix
* add a “deprecation note” in the migration doc that with the next CAPI minor release (v1.3) this func will return the values without suffix

I think this func is barely used by anyone: https://grep.app/search?q=GetIPFamily&filter[lang][0]=Go

File: internal/controllers/topology/cluster/patches/variables/variables.go [hunk 15]
```diff
@@ -213,3 +245,11 @@ func setVariable(variables VariableMap, name string, value interface{}) error {
...
 	return nil
 }
+
+func toJSONCompact(value string) apiextensionsv1.JSON {
+	var compactValue bytes.Buffer
+	if err := json.Compact(&compactValue, []byte(value)); err != nil {
+		panic(err)
+	}
+	return apiextensionsv1.JSON{Raw: compactValue.Bytes()}
+}
```
Comment (fabriziopandini) [inline internal/controllers/topology/cluster/patches/variables/variables.go:255]: I assume there will be only one of this func (either one or the one from @sbueringer PR)

**Review by fabriziopandini — COMMENTED** [review]
@killianmuldoon can we add the new built-in variables to the documentation?

File: internal/controllers/topology/cluster/patches/variables/variables_test.go (hunk 13 above)
Comment (fabriziopandini) [reply internal/controllers/topology/cluster/patches/variables/variables_test.go]: Given that we document IPFamily as 
// IPFamily is the IPFamily the Cluster is operating in. One of Invalid, IPv4, IPv6, DualStack.

And that we have to backport, I would simply drop the suffix

File: internal/controllers/topology/cluster/patches/variables/variables.go (hunk 15 above)
Comment (killianmuldoon) [reply internal/controllers/topology/cluster/patches/variables/variables.go:255]: Yeah - just the depending on which is merged first I guess.

Comment (killianmuldoon) [reply internal/controllers/topology/cluster/patches/variables/variables.go:255]: I've added them in "writing a clusterclass" is there somewhere else they should go?

File: internal/controllers/topology/cluster/patches/variables/variables_test.go (hunk 13 above)
Comment (killianmuldoon) [reply internal/controllers/topology/cluster/patches/variables/variables_test.go]: @fabriziopandini just to be clear - I should drop the suffix in the existing function?

File: internal/controllers/topology/cluster/patches/variables/variables.go (hunk 15 above)
Comment (sbueringer) [reply internal/controllers/topology/cluster/patches/variables/variables.go:255]: And in the proposal. (you can check my PR)

File: internal/controllers/topology/cluster/patches/variables/variables_test.go (hunk 13 above)
Comment (fabriziopandini) [reply internal/controllers/topology/cluster/patches/variables/variables_test.go]: The API cannot be changed, otherwise it will be a breaking change.
However, we can drop the suffix generated by the API when setting the built-in variable

**Comment by fabriziopandini** [comment]
lgtm pending rebase on top of 6208 and removing IPFamily suffix from the built-in variable value

File: internal/controllers/topology/cluster/patches/variables/variables_test.go (hunk 13 above)
Comment (killianmuldoon) [reply internal/controllers/topology/cluster/patches/variables/variables_test.go]: So like this: https://github.com/kubernetes-sigs/cluster-api/blob/8dec3d73b743c36690474d096e9c99330f227bb8/api/v1alpha4/cluster_types.go#L384

Just updated now :smile:

File: api/v1alpha4/cluster_types.go [hunk 16]
```diff
@@ -381,7 +381,7 @@ const (
 )
 
 func (f ClusterIPFamily) String() string {
-	return [...]string{"InvalidIPFamily", "IPv4IPFamily", "IPv6IPFamily", "DualStackIPFamily"}[f]
+	return [...]string{"Invalid", "IPv4", "IPv6", "DualStack"}[f]
```
Comment (sbueringer) [inline api/v1alpha4/cluster_types.go:384]: I think Fabrizio meant transforming the strings returned by the func in the variable code (or alternatively something with switch case, which is probably more stable)

Comment (killianmuldoon) [reply api/v1alpha4/cluster_types.go:384]: https://github.com/kubernetes-sigs/cluster-api/pull/6171#discussion_r816074839 :+1:

Comment (fabriziopandini) [reply api/v1alpha4/cluster_types.go:384]: This is a breaking change.
I was thinking to drop the suffix only when setting the builtin variable

Comment (sbueringer) [reply api/v1alpha4/cluster_types.go:384]: I'll continue above

File: internal/controllers/topology/cluster/patches/variables/variables_test.go (hunk 13 above)
Comment (sbueringer) [reply internal/controllers/topology/cluster/patches/variables/variables_test.go]: I think Fabrizio meant transforming the strings returned by the func in the variable code (or alternatively something with switch case, which is probably more stable)

This change is currently a breaking change. (and in the v1alpha4 package :) )

Comment (killianmuldoon) [reply internal/controllers/topology/cluster/patches/variables/variables_test.go]: Upadated, but I think we might be better off deprecating and adding the method in the API instead of creating 2 .toString methods for this enum.

Comment (fabriziopandini) [reply internal/controllers/topology/cluster/patches/variables/variables_test.go]: @killianmuldoon thanks for updating
we can tackle API deprecation in a separated PR, but I would like to keep this as small as possible for enabling backporting

Comment (killianmuldoon) [reply internal/controllers/topology/cluster/patches/variables/variables_test.go]: Sure thing! I'll open an additional PR and we can discuss there.

File: internal/controllers/topology/cluster/patches/variables/variables_test.go [hunk 17]
```diff
@@ -416,12 +530,3 @@ func TestMachineDeployment(t *testing.T) {
 func toJSON(value string) apiextensionsv1.JSON {
 	return apiextensionsv1.JSON{Raw: []byte(value)}
 }
-
```
Comment (sbueringer) [inline internal/controllers/topology/cluster/patches/variables/variables_test.go:419]: Let's keep this func in the test file. I think it's only used in tests

File: internal/controllers/topology/cluster/patches/variables/variables.go [hunk 18]
```diff
@@ -176,6 +192,22 @@ func Global(clusterTopology *clusterv1.Topology, cluster *clusterv1.Cluster) (Va
 			},
 		},
 	}
+	if cluster.Spec.ClusterNetwork != nil {
+		clusterNetworkIPFamily, err := cluster.GetIPFamily()
+		if err != nil {
+			return nil, err
+		}
+		builtin.Cluster.Network = &ClusterNetworkBuiltins{
+			ServiceDomain: &cluster.Spec.ClusterNetwork.ServiceDomain,
```
Comment (sbueringer) [inline internal/controllers/topology/cluster/patches/variables/variables.go:201]: Missed it before. Can we please only set ServiceDomain if the string is != "". 

Currently when serviceDomain is not set in YAML it leads to an empty `cluster.Spec.ClusterNetwork.ServiceDomain` field which then leads to a builtin variable set to an empty string.

Comment (sbueringer) [reply internal/controllers/topology/cluster/patches/variables/variables.go:201]: (You can reproduce via a new table test case which has ClusterNetwork set, but not ServiceDomain)

**Review by sbueringer — COMMENTED** [review]
Sorry last nits, otherwise lgtm

File: internal/controllers/topology/cluster/patches/variables/variables_test.go (hunk 13 above)
Comment (sbueringer) [reply internal/controllers/topology/cluster/patches/variables/variables_test.go]: Yup good point, we can't really cherry-pick deprecations.

File: internal/controllers/topology/cluster/patches/variables/variables_test.go (hunk 17 above)
Comment (killianmuldoon) [reply internal/controllers/topology/cluster/patches/variables/variables_test.go:419]: Is this about toJsonCompact?

Comment (sbueringer) [reply internal/controllers/topology/cluster/patches/variables/variables_test.go:419]: Yes.

Comment (sbueringer) [reply internal/controllers/topology/cluster/patches/variables/variables_test.go:419]: (commented on the left side of the diff view in "files")

**Comment by sbueringer** [comment]
Thank you!!

/lgtm
/cherry-pick release-1.1

**Comment by sbueringer** [comment]
@killianmuldoon Can you please create a cherry-pick PR manually? Thank you!
